project/src/components/review-form/review-form.tsx

```diff
@@ -1,66 +1,53 @@
-import { FormEvent, useState } from 'react';
+import { ChangeEvent, FormEvent, Fragment, useState } from 'react';
+import { useDispatch } from 'react-redux';
+import { useParams } from 'react-router-dom';
+import { postCommentAction } from '../../store/api-actions';
+import { RATINGS } from '../../const';
 
 function ReviewForm(): JSX.Element {
-  const [review, setReview] = useState('');
+  const [review, setReview] = useState({comment: '', rating: 0});
   const [isCompletedReview, setIsCompletedReview] = useState(false);
+  const dispatch = useDispatch();
+  const {id} = useParams<{id: string}>();
 
   const handleSubmit = (e: FormEvent) => {
     e.preventDefault();
-    // eslint-disable-next-line no-alert
-    alert(`Your review: ${review}`);
-    setReview('');
+    dispatch(postCommentAction(review, Number(id)));
```

Comment:
```suggestion
    dispatch(postCommentAction(review, parseInt(id)));
```
Почитай думаю так будет код немного стабильнее ? https://thisthat.dev/number-constructor-vs-parse-int/